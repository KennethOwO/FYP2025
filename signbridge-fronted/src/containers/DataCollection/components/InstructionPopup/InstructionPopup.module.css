.noScroll {
    overflow: hidden;
}

.instruction-popup {
    width: 1000px;
    max-height: 85vh;
    background-color: #f9f9f9;
    border: 1px solid #ccc;
    border-radius: 15px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    font-size: 18px;
    color: #333;
    padding: 30px;
    z-index: 1001;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: none;
    overflow-y: auto;

    /* Hide scrollbar */
    -ms-overflow-style: none; /* IE and Edge */
    scrollbar-width: none; /* Firefox */

    &::-webkit-scrollbar {
        display: none; /* Chrome, Safari and Opera */
    }
}

.instruction-popup.shown {
    display: block;
}

.instruction_popup_header {
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    padding: 15px 0;
    margin-bottom: 15px;
}

.instruction_title {
    margin: 0;
    text-align: center;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    font-size: 32px;
}

.instruction_popup_details {
    height: 90%;
    overflow-y: auto;
    display: flex;
    flex-direction: column;
    margin-top: 10px;
}

.line_separator {
    width: 95%;
    height: 1px;
    background-color: #ccc;
    margin: 20px 0;
}

.instruction_popup_details_section1 {
    background-color: rgb(235, 235, 235);
}

.instruction_popup_details_section1 p {
    font-size: 16px;
    padding: 10px;
}

.instruction_popup_details_section2 {
    width: 100%;
}

.instruction_note {
    margin-top: 20px;
}

.instruction_popup_content_wrapper {
    display: flex;
}

.instruction_popup_content {
    flex: 1;
    max-width: 100%;
    margin-top: 10px;
    justify-content: space-between;
    background-color: rgb(235, 235, 235);
    margin-right: 10px;
}

.instruction_popup_details_section2 p {
    font-size: 26px;
    font-weight: bold;
}

.instruction_popup_content ul {
    list-style-type: disc;
    padding-left: 30px;
    flex: 1;
    margin-right: 20px;
}

.instruction_popup_content li {
    font-size: 16px;
    padding: 10px 0;
}

.instruction_popup_details_image {
    flex: 1;
    max-width: 40%;
    margin-top: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.instruction_image {
    width: 100%;
    height: auto;
    object-fit: contain;
}

.fa {
    position: absolute;
    right: 0;
    top: 50%;
    transform: translateY(-50%);
    font-size: 36px;
    cursor: pointer;
    z-index: 1;
}

.fa:hover {
    text-shadow: 0 4px 8px rgba(0, 0, 0, 0.5);
}

/* For devices with a max-width of 480px (small screens) */
@media (min-width: 320px) and (max-width: 480px) {
    .instruction-popup {
        width: 90%;
        /* Full width on small screens */
        height: auto;
        /* Auto height for flexibility */
        padding: 15px;
        /* Reduced padding */
        font-size: 16px;
        /* Smaller font size for readability */
    }

    .instruction_popup_header {
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
        width: 100%;
        position: relative;
        padding: 15px 0;
        margin-bottom: 15px;
    }

    .instruction_title {
        font-size: 25px;
        margin: 0;
    }

    .fa {
        font-size: 30px;
        position: absolute;
        right: 0;
        top: 50%;
        transform: translateY(-50%);
    }

    .instruction_popup_details {
        flex-direction: column;
        /* Stack details vertically */
    }

    .instruction_popup_content_wrapper {
        flex-direction: column;
        align-items: center;
        margin-top: 10px;
    }

    .instruction_popup_details_image {
        max-width: 95%;
        margin: 20px auto;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .instruction_image {
        width: 100%;
    }
}

/* For devices with a width between 481px and 640px (mobile) */
@media (min-width: 481px) and (max-width: 640px) {
    .instruction-popup {
        width: 95%;
        /* Nearly full width on mobile */
        padding: 20px;
        /* Slightly more padding */
    }

    .instruction_popup_content {
        padding-right: 10px;
        /* Adjust padding for list */
    }

    .instruction_popup_content_wrapper {
        flex-direction: column;
        align-items: center;
        margin-top: 10px;
        padding: 10px;
    }

    .instruction_popup_details_image {
        max-width: 90%;
        margin: 20px auto;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .instruction_image {
        width: 100%;
    }
}

/* For devices with a width between 641px and 768px (tablets and small devices) */
@media (min-width: 641px) and (max-width: 768px) {
    .instruction-popup {
        width: 90%;
        /* Set width to 90% for tablets */
    }

    .instruction_popup_content {
        padding-right: 15px;
        /* Slight padding adjustment */
    }

    .instruction_popup_content_wrapper {
        flex-direction: column;
        /* Keep the list and image side by side */
    }

    .instruction_popup_details_image {
        max-width: 90%;
        margin: 20px auto;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .instruction_image {
        width: 100%;
        /* Ensure image fills its container */
    }
}

/* For devices with a width between 769px and 1024px (tablets) */
@media (min-width: 769px) and (max-width: 1024px) {
    .instruction-popup {
        width: 80%;
        /* Set width to 80% for larger tablets */
    }

    .instruction_popup_content {
        padding-right: 20px;
        /* More padding for larger space */
    }

    .instruction_popup_content_wrapper {
        flex-direction: row;
        /* Keep side by side layout */
    }

    .instruction_popup_details_image {
        max-width: 40%;
        /* Limit image width */
    }
}

/* For devices with a width between 1025px and 1280px (small desktops) */
@media (min-width: 1025px) and (max-width: 1280px) {
    .instruction-popup {
        width: 70%;
        /* Set width to 70% for small desktops */
        height: auto;
        /* Allow auto height */
    }

    .instruction_popup_details {
        flex-direction: column;
        /* Keep details stacked vertically */
    }
}

/* For devices with a width between 1281px and 1440px (larger desktops) */
@media (min-width: 1281px) and (max-width: 1440px) {
    .instruction-popup {
        width: 60%;
        /* Adjust width for larger desktops */
        height: auto;
        /* Allow auto height */
    }
}
