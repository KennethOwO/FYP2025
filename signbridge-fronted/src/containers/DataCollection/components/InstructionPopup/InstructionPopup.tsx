import React, { useEffect, useRef } from "react";
import styles from "./InstructionPopup.module.css";
import InstructionImage from "/images/Aaron-demo.jpg";
import { useTranslation } from "react-i18next";

interface InstructionPopupProps {
  showInstructionPopup: boolean;
  onClose: () => void;
}

const InstructionPopup: React.FC<InstructionPopupProps> = ({
  showInstructionPopup,
  onClose,
}) => {
  const { t, i18n } = useTranslation();
  const popupRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        popupRef.current &&
        !popupRef.current.contains(event.target as Node)
      ) {
        onClose();
      }
    };

    if (showInstructionPopup) {
      document.addEventListener("mousedown", handleClickOutside);
      document.body.classList.add(styles.noScroll);
    } else {
      document.body.classList.remove(styles.noScroll);
    }

    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
      document.body.classList.remove(styles.noScroll);
    };
  }, [showInstructionPopup, onClose]);

  return (
    <div
      className={`${styles["instruction-popup"]} ${
        showInstructionPopup ? styles["shown"] : ""
      }`}
      ref={popupRef}
    >
      <div className={styles.instruction_popup_header}>
        <h1 className={styles.instruction_title}>{t("instructions")}</h1>
        <i
          className={`${styles.fa} fa fa-close`}
          data-testid="click-close-btn"
          onClick={onClose}
        ></i>
      </div>
      <div className={styles.instruction_popup_details}>
        <div className={styles.instruction_popup_details_section1}>
          <p>
            <strong>1.</strong> {t("inslogin")}
          </p>
          <p>
            <strong>2.</strong> {t("insFillDetails")}
          </p>
          <p>
            <strong>3.</strong> {t("insUploadDemoVideo")}
          </p>
          <p>
            <strong>4.</strong> {t("insResetBtn")}
          </p>
        </div>
        {/* <hr className={styles.line_separator} /> */}
        <div className={styles.instruction_popup_details_section2}>
          <p className={styles.instruction_note}>{t("insNote")}</p>
          <div className={styles.instruction_popup_content_wrapper}>
            <div className={styles.instruction_popup_content}>
              <ul>
                <li>{t("insVideoBg")}</li>
                <li>{t("insVideoMp4")}</li>
                <li>{t("insVideoBody")}</li>
                <li>{t("insVideoPurpose")}</li>
              </ul>
            </div>
            <div className={styles.instruction_popup_details_image}>
              <img
                src={InstructionImage}
                alt="instruction-image"
                className={styles.instruction_image}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default InstructionPopup;
