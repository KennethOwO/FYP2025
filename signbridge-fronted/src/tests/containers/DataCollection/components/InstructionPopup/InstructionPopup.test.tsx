import React from "react";
import InstructionPopup from "../../../../../containers/DataCollection/components/InstructionPopup/InstructionPopup";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";

// Mock the translation functions
jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

describe("Test InstructionPopup", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  const props = {
    showInstructionPopup: true,
    onClose: jest.fn(),
  };
  it("should render correctly", () => {
    const tree = create(<InstructionPopup {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
