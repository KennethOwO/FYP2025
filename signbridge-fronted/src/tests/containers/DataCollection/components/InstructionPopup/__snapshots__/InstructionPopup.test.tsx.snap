// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InstructionPopup should render correctly 1`] = `
<div
  className="undefined undefined"
>
  <div>
    <h1>
      instructions
    </h1>
    <i
      className="undefined fa fa-close"
      data-testid="click-close-btn"
      onClick={[MockFunction]}
    />
  </div>
  <div>
    <div>
      <p>
        <strong>
          1.
        </strong>
         
        inslogin
      </p>
      <p>
        <strong>
          2.
        </strong>
         
        insFillDetails
      </p>
      <p>
        <strong>
          3.
        </strong>
         
        insUploadDemoVideo
      </p>
      <p>
        <strong>
          4.
        </strong>
         
        insResetBtn
      </p>
    </div>
    <hr />
    <div>
      <p>
        insNote
      </p>
      <div>
        <ul>
          <li>
            insVideoBg
          </li>
          <li>
            insVideoMp4
          </li>
          <li>
            insVideoBody
          </li>
          <li>
            insVideoPurpose
          </li>
        </ul>
      </div>
    </div>
    <div>
      <p>
        insReferImage
      </p>
      <div>
        <img
          alt="instruction-image"
          src="default"
        />
      </div>
    </div>
  </div>
</div>
`;
